### DISCUSSION TITLE
Add a hook/event to ConfigService::get to transform return value

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#9536] [Ideas]
> Created by: Timothy Kelty (timkelty@gmail.com) on 2015/04/02 12:33:05 +0000
> Votes at time of UserVoice import: 1

-----------------------------------

This would make it really easy to write a dotenv, or get your config from anywhere you like.

**Comment by timkelty** [comment]
What I meant to say was - It would make it possible to write a plugin that automatically checked dotenv for values, anytime a config value was accessed.

In Craft 3 the hook or event would probably be in: `craft\services\Config::getConfigSettings`

**Comment by khalwat** [comment]
So... I think this should be solved via https://github.com/craftcms/cms/issues/1607

If the `.env` is optimized into the global configuration, you can access it anywhere.
